packages/stress/src/mode/chat/root_chat.ts

```diff
@@ -126,6 +131,57 @@ export async function startStressChat(opts: {
         )
     })
 
+    logger.info('createGdm')
+    const createGdmResults = await Promise.allSettled(
+        clients
+            .filter(() => probability(chatConfig.gdmProbability))
+            .map((client) => {
+                const randomClients = getRandomClients(clients, 4)
+                const memberIds = randomClients.map((c) => c.userId)
+                return client.agent.gdms
+                    .createGDM(memberIds)
+                    .then(({ streamId }) => {
+                        logStep(client, 'CREATE_GDM', true, { streamId, memberIds })
+                        return { streamId, randomClients }
+                    })
+                    .catch((e) => {
+                        logStep(client, 'CREATE_GDM', false, { reason: e })
+                        throw e
+                    })
+            }),
+    ).then((results) => {
+        return results.flatMap((r) => {
+            if (r.status === 'rejected') {
+                errors.push(r.reason)
+            }
+            if (r.status === 'fulfilled') {
+                return r.value
+            }
+            return { streamId: undefined, randomClients: undefined }
+        })
+    })
+
+    logger.info('gdmChat')
+    await Promise.all(
+        createGdmResults.map(({ streamId, randomClients }) => {
+            if (!streamId || !randomClients) {
+                return
+            }
+            const promises = randomClients.map((client) =>
+                gdmChat(client, streamId, chatConfig)
+                    .then(({ eventId }) => {
+                        logStep(client, 'GDM_CHAT', true, { streamId, eventId })
+                    })
+                    .catch((e) => {
+                        errors.push(e)
+                        logStep(client, 'GDM_CHAT', false, { reason: e })
+                        throw e
+                    }),
+            )
+            return Promise.allSettled(promises)
+        }),
+    )
```

Comment:
this only creates gdms with local clients... but, that's fine for this iteration i guess.